std/array.d

```diff
@@ -3661,7 +3672,7 @@ unittest
     {
         return [new shared A].inputRangeObject;
     }
-    auto res = foo.array;
+    static assert(__traits(compiles, foo.array));
```

Comment:
I didn't remove any traits compiles in this PR. The line I deleted here was declaring res just to make sure that foo.array compiles. Nesting it with a traits compiles gets rid of the unused variable while preserving functionality